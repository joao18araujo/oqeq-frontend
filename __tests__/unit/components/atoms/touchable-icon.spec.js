import React from 'react';
import { shallow } from 'enzyme';
import { TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import TouchableIcon from '~components/atoms/touchable-icon';

const initialProps = (props) => ({
  icon: 'arrow-left',
  onPress: () => {},
  size: 10,
  color: '#0080ff',
  ...props,
});

describe('TouchableIcon', () => {
  const onPressMock = jest.fn();
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallow(
      <TouchableIcon {...initialProps({ onPress: onPressMock })} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should pass onPress to TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity).props()).toEqual(
      expect.objectContaining({
        onPress: onPressMock,
      })
    );
  });

  it('should pass icon, size and color to Icon', () => {
    expect(wrapper.find(Icon).props()).toEqual(
      expect.objectContaining({
        name: 'arrow-left',
        size: 10,
        color: '#0080ff',
      })
    );
  });
});
