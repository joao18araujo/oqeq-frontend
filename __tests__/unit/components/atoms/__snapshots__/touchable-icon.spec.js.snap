// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchableIcon should match the snapshot 1`] = `"<View style=\\"opacity:1\\" focusable=\\"true\\"><Text style=\\"0:[object Object];2:[object Object];3:[object Object]\\">ÔÅ†</Text></View>"`;
